tools/ai_sidecar/server.py

```
@@ -276,8 +492,21 @@ def main() -> None:
         _run_compare_laps(args.compare_laps[0], args.compare_laps[1])
         return
     reply = not args.no_reply
+
+    if args.external_bind is not None:
+        host = args.external_bind
+        if not _is_loopback(host) and not args.token:
+            raise SystemExit(
+                "--external-bind requires --token for non-loopback addresses "
+                "(refusing to expose unauthenticated socket)"
+            )
```

Ruff (0.15.10)

[warning] Avoid specifying long messages outside the exception class

(TRY003)